### PR TITLE
Code Cleanups

### PR DESCRIPTION
**Pull request by enzian**
a load of code cleanups in order to address #115

### CODE REVIEW

**Comment by enzian** [comment]
WIP

**Comment by enzian** [comment]
WIP

**Comment by enzian** [comment]
I signed it

**Comment by enzian** [comment]
I signed it

File: build/Version.props [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-  <PropertyGroup>
-    <!-- 
-      Semantic Version. See http://semver.org for full details. 
-      Update for every public release. 
-    -->
-    <SemanticVersionMajor>0</SemanticVersionMajor>
-    <SemanticVersionMinor>1</SemanticVersionMinor>
-    <SemanticVersionPatch>0</SemanticVersionPatch>
-    <!--Valid values: beta1, beta2, EMPTY for stable -->
-    <PreReleaseMilestone>alpha</PreReleaseMilestone>
```
Comment (lmolkova) [inline build/Version.props]: I assume minVer does some versioning magic, but we don't seem to set these versions anywhere. So can you help me understand what our versioning story would be with this change?

Comment (lmolkova) [reply build/Version.props]: I saw you created https://github.com/open-telemetry/opentelemetry-dotnet/issues/130. Can you please separate it from code cleanups and submit another PR or have a discussion in the #130. 

My concern is that we do not yet have tags in place and I want to make sure build scripts will work nicely with it even before we have tags.

But it shouldn't stop cleanups from merging

Comment (enzian) [reply build/Version.props]: > Can you please separate it from code cleanups and submit another PR or have a discussion in the #130.

I would love to do that but there were actually quite a few warnings about the way version were defined now, so I thought I'll just go ahead with this change now! 

Minver should be working only with the references to the package set as they are now, we just need to push some sensible tag for it to detect something apart from 0.0.0-alpha.0.offset...

**Comment by enzian** [comment]
@lmolkova thanks for the feedback but before we merge this we need to talk about a few rules that I disabled on my own recognizance, in particular **CS1591**.

**CS1591** disables warnings for non-documented public members and types. This rules has been questioned here before. I'd suggest to only enable this rule for `OpenTelemetry` and `OpenTelemetry.Abstractions` since all other project are basically contrib projects, most of which have been adopted from the open census project directly and do not conform to the same level of code quality that the opentelemetry libraries do! Enabling this rule will force me (the new guy at the party) to write docs for APIs I understand at most partially which does not lead to great documentation

Are there any other rules we need to revisit?

**Comment by enzian** [comment]
also `TreatWarningsAsErrors` is not enabled yet ;-)

**Comment by lmolkova** [comment]
@SergeyKanzhelev @austinlparker could you please merge?

**Pull request merged**